podcasts/Sharing/Clip/VideoExporter.swift

```diff
@@ -0,0 +1,203 @@
+                              videoWriterInput: videoWriterInput,
+                              videoWriter: videoWriter,
+                              adaptor: adaptor,
+                              progress: progress,
+                              frameCount: frameCount)
+    }
+
+    // Part of Step 1
+    private static func writeFrames<Content: AnimatableContent>(of view: Content, size: CGSize, fps: Int, videoWriterInput: AVAssetWriterInput, videoWriter: AVAssetWriter, adaptor: AVAssetWriterInputPixelBufferAdaptor, progress: Progress, frameCount: Int) async throws {
+        let counter = Counter()
+        try await videoWriterInput.unsafeRequestMediaDataWhenReady { continuation in
+            while await counter.count <= frameCount, videoWriterInput.isReadyForMoreMediaData {
+                do {
+                    try await counter.run {
+                        let frameProgress = Double(await counter.count) / Double(frameCount)
+                        view.update(for: frameProgress)
+
+                        let buffer = try await self.pixelBuffer(for: view, size: size)
+                        let frameTime = CMTime(seconds: Double(await counter.count) / Double(fps), preferredTimescale: CMTimeScale(NSEC_PER_SEC))
+                        adaptor.append(buffer.wrappedValue, withPresentationTime: frameTime)
+                        progress.completedUnitCount += 1
+                    }
+                } catch {
+                    continuation.resume(throwing: error)
+                    return
+                }
+            }
+
+            if await counter.count >= frameCount {
+                videoWriterInput.markAsFinished()
+                videoWriter.finishWriting {
+                    continuation.resume()
+                }
+            }
+        }
+    }
+
+    @available(iOS 16, *)
+    @MainActor
+    private static func pixelBuffer(for view: some View, size: CGSize) throws -> UnsafeTransfer<CVPixelBuffer> {
+        try UnsafeTransfer(view.frame(width: size.width, height: size.height).pixelBuffer(size: size))
+    }
+
+    // Part 2 of video export, creating the final track from the initial video loop
+    private static func createFinalComposition(from sourceURL: URL, with parameters: Parameters, to outputURL: URL, progress: Progress) async throws -> AVComposition {
+        let asset = AVAsset(url: sourceURL)
+        let composition = AVMutableComposition()
+
+        guard let compositionVideoTrack = composition.addMutableTrack(withMediaType: .video, preferredTrackID: kCMPersistentTrackID_Invalid),
+              let sourceVideoTrack = try? await asset.loadTracks(withMediaType: .video).first else {
+            throw ExportError.failedToCreateCompositionTrack
+        }
+
+        let sourceTimeRange = try await asset.load(.duration)
+        var currentTime: CMTime = .zero
+
+        while currentTime < CMTime(seconds: parameters.duration, preferredTimescale: 600) {
```

Comment:
Loops the video track over the full duration. Since our animation is a looping background change, we can just reuse those frames instead of generating frames over the whole duration, dramatically speeds up export especially on long clips. It rarely took more than 45 seconds to export in my testing regardless of clip length.